frontend/src/settings/Preferences.jsx

```diff
@@ -1,16 +1,89 @@
 import * as React from 'react';
 import {
-  Text,
+  Alert, Button, Switch, Platform,
 } from 'react-native';
+import { LinearGradient } from 'expo-linear-gradient';
+import RNDateTimePicker from '@react-native-community/datetimepicker';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useFocusEffect } from '@react-navigation/native';
 import {
-  BoilerplateView,
+  PreferencesView, CardTitle, PreferencesTitle,
 } from '../Themes';
 
 function Preferences() {
+  const [darkMode, setDarkMode] = React.useState(false);
+  const [Alarm, setAlarm] = React.useState(true);
+  const [twelveHour, setTwelveHour] = React.useState(true);
+  const [alarmTime, setAlarmTime] = React.useState(new Date());
+  const [show, setShow] = React.useState(false);
+
+  const onChange = (event, selectedDate) => {
+    const currentDate = selectedDate || alarmTime;
+    setShow(Platform.OS === 'ios');
+    setAlarmTime(currentDate);
+    Alert.alert(`Alarm set to: ${currentDate.toTimeString().split(' ')[0]}`);
```

Comment:
It's probably not a good idea to have an alert pop-up every single time the value changes. If the time is at like say 8:00 AM and I start scrolling the minutes, every time my finger lifts off I get an alert, even tho maybe I'm not yet at the time I want. Or maybe I change the minutes first and then go to change the hours but I have to dismiss the alert every time.

Reply:
agreed, I'll fix that.